gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDatasetTest.java

```diff
@@ -216,6 +223,40 @@ public void testGetFilesAtPath() throws IOException {
       Assert.assertTrue(candidateFiles.contains(PathUtils.getPathWithoutSchemeAndAuthority(fileStatus.getPath()).toString()));
     }
 
+    // test ds of daily/yyyy-MM-dd-HH-mm-ss
+    datePattern = "yyyy-MM-dd-HH-mm-ss";
+    formatter = DateTimeFormat.forPattern(datePattern);
+    endDate = LocalDateTime.now(DateTimeZone.forID(TimeAwareRecursiveCopyableDataset.DEFAULT_DATE_PATTERN_TIMEZONE));
+
+    candidateFiles = new HashSet<>();
+    for (int i = 0; i < MAX_NUM_DAILY_DIRS; i++) {
+      String startDate = endDate.minusDays(i).withMinuteOfHour(random.nextInt(60)).withSecondOfMinute(random.nextInt(60)).toString(formatter);
```

Comment:
It shouldn't affect the test since the test is looking for the past 2 days lookback, so the minute/second shouldn't affect the outcome. We just want to make sure that the date partitioned by the second will actually be picked up (it wasn't before due to how the iterator would assume every second path exists, but only look for paths in the increment of minutes)

Reply:
ah, yes.  makes sense--good use for `random`